src/Session.php

```diff
@@ -359,7 +359,7 @@ public function query($yql, array $parameters = null)
         $query = $this->newQuery($yql)
             ->parameters($parameters)
             ->txControl($tx_control)
-            ->keepInCache($this->keep_query_in_cache);
+            ->keepInCache($this->keep_query_in_cache ?? ($parameters&&count($parameters)>0));
```

Comment:
"??" is synonim for isset function, it doesn't check for null

at line 71 you initialize $this->keep_query_in_cache as null. It mean - right part of expression will not execute.